tests/virt_autotest/sriov_network_card_pci_passthrough.pm

```diff
@@ -132,13 +135,13 @@ sub run_test {
             assert_script_run("virsh nodedev-reattach $vfs[$i]->{host_id}", 60);
             record_info("Reattach VF to host", "vm=$guest \nvf=$vfs[$i]->{host_id}");
             save_network_device_status_logs($log_dir, $guest, $passthru_vf_count + 4 + $i . "-after_hot_unplug_$vfs[$i]->{host_id}");
+            script_run("ssh root\@$guest 'dmesg' >> $log_dir/dmesg_$guest 2>&1", die_on_timeout => 0) if $i == $passthru_vf_count - 1;
         }
         script_run "lspci | grep Ethernet";
         save_screenshot;
 
         #check host and guest to make sure they work well
-        check_host();
-        check_guest($guest);
+        check_guest_health($guest);
```

Comment:
Would you please share why you prefer/implement in below way?
- check_host_health in post_fail_hook, but no check_guest_health there
- there is already check_guest_health in validate_system_health, but several more such checks are added within sriov_network_card_pci_passthrough.pm

What's  the benefits? 

Besides do you have some summary about the cost(extra time) for adding overall check of host and guest health?

Reply:
> Would you please share why you prefer/implement in below way?
> 
> * check_host_health in post_fail_hook, but no check_guest_health there

- Because post_fail_hook does not which guests to check. Much more effort is needed to do if you make all guests be check-able, such as list all guests, start these guests, check if they are pingable and ssh accessiable, if you think it is quite necessary. 
- Guests may not be installed successfully yet.

> * there is already check_guest_health in validate_system_health, but several more such checks are added within sriov_network_card_pci_passthrough.pm
> What's the benefits?

Because each step may result in coredumps/calltrace in sriove test. For example, that coredump failure [bsc#1204727](https://bugzilla.suse.com/show_bug.cgi?id=1204727) was found in guest just after the interface was brought up in my manual reproduction. but finally after closer look, it was also found in host. I personally like to add more checks to locate bugs as exactly as I can since it does not cost lot.

> Besides do you have some summary about the cost(extra time) for adding overall check of host and guest health?

- The new added validate_system_health test module took 2 minutes around in our product testing. 
- Each checks before each test or in post_fail_hook took 7-10 seconds on hosts with/without calltrace. So for an entire prj1 test, it took 2-4 minutes.

Reply:
Thanks, I am fine with multiple checks to precisely locate where produces the coredump/trace. Glad to know that the cost of this check is low.